### PR TITLE
Chapter7 8 9

### PR DESCRIPTION
**Pull request by SergioGueruiz**
### Hello!
This PR contains exercises from chapter 7, 8 and 9

1. Create `ranges` and calls methods
2. Many ways to create an `Array`
3. Intro to for loop
4. Iterations with while and until
5. Map method

### CODE REVIEW

File: Chapter9/map.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#map method
+#mao and collect are the same
+
+numbers = (1..5).to_a
+squares = numbers.map { |number| number ** 2 }
```
Comment (mauriciosabogal) [inline Chapter9/map.rb:5]: What is the difference between `numbers.map { ... }` and `numbers.each { ... }`?

File: Chapter8/arrayMethod.rb [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+p a == c #False
+p a == d #True
+p c == c2 #False
+
+#It's true if is exactly the same (order, amount, type)
+puts
+#Spaceship operator
+#Returns one of four values, 0, 1, -1 and nil
+
+p 5 <=> 5 #0, if both are equals
```
Comment (mauriciosabogal) [inline Chapter8/arrayMethod.rb:118]: How can you define a function that does the same as this operator `<=>`?

File: Chapter9/keyword.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+puts
+#next keyword
+numbers = [1, 2, 3, "Hello", 5, 6, 7]
+
+numbers.each do |num|
+  if num.is_a?(Integer)
+    puts "The square of #{num} is #{num ** 2}"
+  else
+    next
```
Comment (mauriciosabogal) [inline Chapter9/keyword.rb:24]: When should you use `next` instead of `break` inside a loop?

File: Chapter9/map.rb (hunk 1 above)
Comment (SergioGueruiz) [reply Chapter9/map.rb:5]: `map`: return a new Array with modified elements that we can stores in a variable.
`each`: we need to create a variable before the loop and use into the loop if we want stores the changes that makes.
 example:
```
square = [ ]
arr = (1..3).to_a
arr.each { | num |  square.push(num**2) }
p square #[1,4,9]
```
If we don't create `square` before:
```
square = arr.each { | num |  num**2 }
p square #[1,2,3]
```

File: Chapter8/arrayMethod.rb (hunk 2 above)
Comment (SergioGueruiz) [reply Chapter8/arrayMethod.rb:118]:
```
def spaceship(num1, num2)
  if num1.class.superclass != num2.class.superclass
    return nil
  elsif num1 == num2
    return 0
  elsif num1 < num2
    return -1
  else num1 > num2
    return 1
  end
end

p spaceship(5,5) #0
p spaceship(5,10) #-1
p spaceship(5,4) #1
p spaceship(4.4, 2) #1
p spaceship("Hi", 2) #nil
```

File: Chapter9/keyword.rb (hunk 3 above)
Comment (SergioGueruiz) [reply Chapter9/keyword.rb:24]: When i want exclude any values but i don't want to finish the loop or if i don't want show a value
example:
```
table = ["Name", "Age", "Lastname", "Color", "Music", "City"]
info = ["Sergio", 21, "Guerrero", "Blue", "Rock", "Manizales"]

table.zip(info).each_with_index do |data, i|
  if data.include?("City")
    next
  else
    puts "#{data[0]}: #{data[1]}"
  end
end
```
Prints all the values except "City" and "Manizales"

**Review by mauriciosabogal — APPROVED** [review]

**Pull request merged**